### ISSUE TITLE
UC Proposal: Sharing data from (decentralized) personal storage

### ISSUE THREAD

**Issue by crtahlin** [KantaraInitiative/consent-receipt-v-next#22]
+++++++++++++++++++++++++++++++++++++++++++++++++++++++
Use Case Name
*(A short name)*

Sharing data from (decentralized) personal storage

+++++++++++++++++++++++++++++++++++++++++++++++++++++++
Scope:
*(The name of the system or business area or business function operated)*

(Decentralized) personal data storage.

+++++++++++++++++++++++++++++++++++++++++++++++++++++++
Context:
*(Text about the goal in its operational context)*

Individual can hold his data in (decentralized & encrypted) personal data storage. A data user can query the individual to share some of that data or ask him to answer some questions to provide some yet non-existent data.

+++++++++++++++++++++++++++++++++++++++++++++++++++++++
Primary Actor:
*(The primary actor name or role)*

Individual (Personal data storage operator)

+++++++++++++++++++++++++++++++++++++++++++++++++++++++
Stakeholders & Interests
*(List of all the stakeholders and their interests in the system operation. Stakeholders include actors that interact with the system and others who do not interact with the system.)*

Individual (Personal data storage operator)
Data user

+++++++++++++++++++++++++++++++++++++++++++++++++++++++
Minimal Guarantees
*(An optional section listing the minimal guarantees)*

+++++++++++++++++++++++++++++++++++++++++++++++++++++++
Success Guarantees
*(An optional section listing the success guarantees)*

+++++++++++++++++++++++++++++++++++++++++++++++++++++++
Preconditions:
*(What must be true before this use case can be triggered)*

Individual most hold an account in personal data storage, known to the data user.

+++++++++++++++++++++++++++++++++++++++++++++++++++++++
Triggers
*(What causes the main scenario to start)*

Data user sends to the Individual a request (consent receipt proposal) for some of his existing or yet non-existing data.

+++++++++++++++++++++++++++++++++++++++++++++++++++++++
Main Success Scenario:
*(A brief list of what the actors do which results in the goal being successfully achieved.)*

1. Individual receives a (Data user private key) signed request to share some of his data which can be viewed as a consent proposal
2. Individual accepts the offfer by signing the proposal with his own private key
3. Consent receipt is saved in decentralized storage, accessible to both parties and the transaction is also recorded to a blockchain
4. The process of sharing data via decentralized storage and/or a form to enter data continues

+++++++++++++++++++++++++++++++++++++++++++++++++++++++
Secondary Success Scenario:
*(A second brief list of what the actors do when things don't go down the main path - includes alternative paths and major failure conditions.)*

+++++++++++++++++++++++++++++++++++++++++++++++++++++++

**Comment by LALeVasseur** [comment]
Proposed Project Name for this:  tbd. 

Crt--is this a fair summary of this proposal  "2nd party proffered consent request & handshake"

**Comment by crtahlin** [comment]
@LALeVasseur Yes, I think so. If the "handshake" is understood in a loose sense of establishing a relationship (and not necessarily an active connection / stream, which might be initiated later and the requesting party actually gets data from decentralized storage, not connection to the 1st party).
